### PR TITLE
Ensure that N_ returns unicode

### PR DESCRIPTION
**Pull request by jkimbo**
To match `_(` always returning unicode: https://hg.python.org/cpython/file/2.7/Lib/gettext.py#l246

### CODE REVIEW

**Comment by jkimbo** [comment]
cc @Ealdwulf @keberlin

**Comment by Ealdwulf** [comment]
I don't think we want to do that. We want to call N_ on untranslated strings which might end up passed to  things which don't support unicode yet.

**Comment by jkimbo** [comment]
The issue is mixing up `_(` and `N_(`. Since `_(` will always return unicode strings (even if it isn't translated) I think it makes sense to be consistent. Anything that gets passed to either `_(` or `N_(` should be unicode (if they contain unicode characters) or things will blow up anyway.

**Comment by jkimbo** [comment]
@Ealdwulf added the `_` and `N_` stubs as per our conversation today. `N_` will always return unicode when a top level domain is installed.

File: kano_i18n/init.py [hunk 1]
```diff
@@ -48,6 +54,9 @@ def register_domain(domain, locale_dir=None):
         DOMAINS_TO_REGISTER.append(
             (domain, locale_dir)
         )
+        # and register stubs for _ and N_
+        __builtin__.__dict__['_'] = lambda msg: msg
+        __builtin__.__dict__['N_'] = lambda msg: msg
```
Comment (Ealdwulf) [inline kano_i18n/init.py]: Can we make them sanitize the value on the way through (make sure it's ascii)?

Comment (jkimbo) [inline kano_i18n/init.py]: Not sure what the best value for the `errors` param should be:

```
In [14]: u'¡Hola!'.encode('ascii', 'ignore')
Out[14]: 'Hola!'

In [15]: u'¡Hola!'.encode('ascii', 'replace')
Out[15]: '?Hola!'

In [16]: u'¡Hola!'.encode('ascii', 'xmlcharrefreplace')
Out[16]: '&#161;Hola!'
```

@Ealdwulf @keberlin what do you think?

Comment (Ealdwulf) [inline kano_i18n/init.py]: I guess I would go with 'replace'.

Comment (jkimbo) [inline kano_i18n/init.py]: 👍

**Comment by jkimbo** [comment]
Ready for another pass @Ealdwulf

**Comment by Ealdwulf** [comment]
Looks good to me 👍

**Pull request merged**